core/controllers/base.py

```diff
@@ -230,6 +230,9 @@ def dispatch(self):
 
         if self.payload is not None and self.REQUIRE_PAYLOAD_CSRF_CHECK:
             try:
+                if 'signup' in self.request.uri and not self.user_id:
```

Comment:
Add a comment to explain what is going on here.

Reply:
Done.